deepspeed/module_inject/layers.py

```diff
@@ -13,6 +13,65 @@
 from deepspeed.module_inject.tp_shard import get_shard_size, get_shard_size_list
 
 
+class TensorParallelConv2d(nn.Module):
+
+    def __init__(self, conv, rank, world_size, shard_by_oc):
+        super().__init__()
+        self.rank = rank
+        self.world_size = world_size
+        self.shard_by_oc = shard_by_oc
+        self.shard_weights(conv)
+
+    def shard_weights(self, conv):
```

Comment:
Should have some comments to explain the sharding scheme here.   Better with a simple example to help understanding.